### PR TITLE
task5_6: implement auth with passportjs, store data in postgresql

### PR DESCRIPTION
**Pull request by YahorDauhaliuk**

### CODE REVIEW

File: routes/index.js [hunk 1]
```diff
@@ -1,7 +1,13 @@
...
+import signIn from './auth/signIn';
+import facebook from './auth/facebook';
 import users from './users';
+import product from './product';
 
 export default (app) => {
-	app.use('/api/products', product);
+	app.use('/api/signup', signUp);
+	app.use('/api/signin', signIn);
 	app.use('/api/users', users);
```
Comment (abruzgin) [inline routes/index.js:10]: The `isAuth` middleware should be applied here, therefore this middleware will be applicable only for the the routes not bound to authentication but to Products and Users access

Comment (abruzgin) [reply routes/index.js:10]: And also remove the apply of this middleware in other files

Comment (abruzgin) [reply routes/index.js:10]: Just re-read the theory of Express middlewares

File: middlewares/myPassport.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+				await models.user.comparePassword(password, user.password);
+				return cb(null, user, { message: 'Logged In Successfully' });
+			} catch (err) {
+				return cb(false, null, { message: 'Wrong login or password' });
+			}
+		}
+	)
+);
+
+passport.use(
```
Comment (abruzgin) [inline middlewares/myPassport.js:28]: This task is stood before the `passport` involvement, so please rewrite the Token-based auth avoiding `passport`

File: routes/auth/facebook.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import { Router } from 'express';
+import passport from 'passport';
+const router = Router();
+
+router.get('/', passport.authenticate('facebook', { session: false }), (req, res) => {
+	res.status(200).json('success');
+});
+
```
Comment (abruzgin) [inline routes/auth/facebook.js:8]: As I remember, there should also be a `/callback` route for social auth
Does it really work without `callback`?

Comment (YahorDauhaliuk) [reply routes/auth/facebook.js:8]: Its work.
image

But I use the callback. Its placed in the myPassport.js
image

Comment (abruzgin) [reply routes/auth/facebook.js:8]: I didn't mean the `cb` functon but the `/callback` route
That's what I'm about https://github.com/jaredhanson/passport-facebook#authenticate-requests

Comment (YahorDauhaliuk) [reply routes/auth/facebook.js:8]: In my case it is unnecessary. It works perfectly with one route. As I see, callback route need for a redirect.

**Comment by YahorDauhaliuk** [comment]
All remarks fixed.

**Pull request closed without merging**